### ISSUE TITLE
MA0157 conflicts with CS8403

### ISSUE THREAD

**Issue by mjr4077au** [meziantou/Meziantou.Analyzer#1204]
### Version of the Meziantou.Analyzer NuGet package

3.0.115

### Rule Identifier

MA0157

### Target Framework

net472, net8.0

### C# Language version

C# 14

### Description

<img width="1113" height="347" alt="Image" src="https://github.com/user-attachments/assets/7e70f7b2-3f36-4adb-9b04-c48349a24f78" />

### Reproduction Steps

Minimal code:

```c#
code to reproduce the error
```

**Comment by Frulfump** [comment]
Isn't CS8403 telling you to do `private static async...` and MA0157 is telling you to change the name of the method so you don't end(suffix) the name with `Async` i.e `GetAll`?

I didn't find any good docs for CS8403 https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/async-await-errors#async-method-signature-requirements maybe you could link to some?

https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0157.md
> MA0157 - Do not use 'Async' suffix when a method returns IAsyncEnumerable<T>
>
>Methods that do not return IAsyncEnumerable<T> should not have the Async suffix.
>
>// compliant
>IAsyncEnumerable<string> Foo() => throw null;
>
>// non-compliant
>IAsyncEnumerable<string> FooAsync() => throw null;

The docs text is a bit confusing, first it says 
`Do not use 'Async' suffix when a method returns IAsyncEnumerable<T>`
Then it says `Methods that do not return IAsyncEnumerable<T> should not have the Async suffix.
` but shouldn't that say `Methods that return IAsyncEnumerable<T> should not have the Async suffix`?! But then the example shows code that doesn't return anything. 

I think the confusion comes from return type vs what's actually being returned so maybe the docs & analyzer text could be clarified?

What does your code actually return (Based on the message of CS8403 it's returning an  IAsyncEnumerable<SessionInfo> but maybe it's throwing or something instead)?

**Comment by mjr4077au** [comment]
You're quite right, and this one's actually my fault as I forcibly turned on every Meziantou analyser rule and didn't realise the project had rules that juxtaposed with each other:

<img width="849" height="132" alt="Image" src="https://github.com/user-attachments/assets/7bae8020-556c-42b7-bf0b-9f12e30b2be6" />

I've disabled MA0157 as I'd prefer MA0156 here.

**Comment by meziantou** [comment]
> You're quite right, and this one's actually my fault as I forcibly turned on every Meziantou analyser rule and didn't realise the project had rules that juxtaposed with each other:

You may need to tweak rules that are disabled by default. There are reasons for them to be disabled (false-positive, not generally applicable, hard to fix).
The column `Is enabled` in the [documentation](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/README.md) will tell you which rules you should review

**Comment by mjr4077au** [comment]
This was a purposeful endeavour as I feel it's best for my projects that everything is enabled and I make decisions on what I want disabled. I sent this report off without any coffee within me and I think it shows 😅
